src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,188 @@
+# gpkg_contents table.
+function gpkgextract(db, ; layer=1)
+  rowtable = DBInterface.execute(
+    # According to https://www.geopackage.org/spec/#r16 
+    # Values of the gpkg_contents table srs_id column 
+    # SHALL reference values in the gpkg_spatial_ref_sys table srs_id column
+    # According to https://www.geopackage.org/spec/#r18
+    # The gpkg_contents table SHALL contain a row 
+    # with a lowercase data_type column value of "features" 
+    # for each vector features user data table or view.
+    db,
+    """
+    SELECT g.table_name AS tablename, g.column_name AS geomcolumn, 
+    c.srs_id AS srsid, g.z, srs.organization AS org, srs.organization_coordsys_id AS orgcoordsysid,
+    ( SELECT type FROM sqlite_master WHERE lower(name) = lower(c.table_name) AND type IN ('table', 'view')) AS object_type
+    FROM gpkg_geometry_columns g, gpkg_spatial_ref_sys srs
+    JOIN gpkg_contents c ON ( g.table_name = c.table_name )
+    WHERE c.data_type = 'features'
+    AND object_type IS NOT NULL
+    AND g.srs_id = srs.srs_id
+    AND g.srs_id = c.srs_id
+    AND g.z IN (0, 1, 2)
+    AND g.m = 0
+    LIMIT 1 OFFSET ($layer-1);
+    """
+  )
+
+  # Note: first feature table that is read specifies the CRS to be used on all feature tables resulted from SELECT statement
+  firstrow = first(rowtable)
+
+  # According to https://www.geopackage.org/spec/#r33, feature table geometry columns
+  # SHALL contain geometries with the srs_id specified for the column by the gpkg_geometry_columns table srs_id column value.
+  tablename, geomcolumn = firstrow.tablename, firstrow.geomcolumn
+  srsid, org, orgcoordsysid = firstrow.srsid, firstrow.org, firstrow.orgcoordsysid
+
+  # identify coordinate reference system from srsid
+  if iszero(srsid)
+    crs = LatLon{WGS84Latest}
+  elseif srsid != -1
+    if org == "EPSG"
+      crs = CoordRefSystems.get(EPSG{orgcoordsysid})
+    elseif org == "ESRI"
+      crs = CoordRefSystems.get(ERSI{orgcoordsysid})
+    end
+  else # defaults to Cartesian CRS with no datum
+    crs = Cartesian{NoDatum}
+  end
+
+  # According to https://www.geopackage.org/spec/#r14
+  # The table_name column value in a gpkg_contents table row 
+  # SHALL contain the name of a SQLite table or view.
+  tableinfo = SQLite.tableinfo(db, tablename)
+
+  # "pk" (either zero for columns that are not part of the primary key, or the 1-based index of the column within the primary key)
+  columns = [name for (name, pk) in zip(tableinfo.name, tableinfo.pk) if pk == 0]
+  gpkgbinary = DBInterface.execute(db, "SELECT  $(join(columns, ',')) FROM $tablename;")
+  table = map(gpkgbinary) do row
+    # According to https://www.geopackage.org/spec/#r30
+    # A feature table or view SHALL have only one geometry column.
+    geomindex = findfirst(==(Symbol(geomcolumn)), keys(row))
+    rowvals = map(keys(row)[[begin:(geomindex - 1); (geomindex + 1):end]]) do key
+      key, getproperty(row, key)
+    end
```

Comment:
If row is a NamedTuple, you can filter it with
```suggestion
    rowvals = (; (k => v for (k, v) in pairs(row) if k != geomcolumn)...)
```

Reply:
This fails test for duplicate field names 

`  duplicate field name in NamedTuple: "geometry" is not unique`